src/Kernel32.Shared/Kernel32+SafeObjectHandle.cs

```diff
@@ -37,7 +37,7 @@ public SafeObjectHandle(IntPtr preexistingHandle, bool ownsHandle)
             }
 
             /// <inheritdoc />
-            public override bool IsInvalid => this.handle == INVALID_HANDLE_VALUE;
+            public override bool IsInvalid => this.handle == INVALID_HANDLE_VALUE || this.handle == IntPtr.Zero;
```

Comment:
It's due to compatibility with Win16, all functions inherited from this age (Like `OpenProcess`) return `NULL` instead of `INVALID_HANDLE_VALUE`. Raymon Chen got a good explanation : http://blogs.msdn.com/b/oldnewthing/archive/2004/03/02/82639.aspx

> Second, it means that if you write a generic handle-wrapping class, you have to be mindful of two possible "not a handle" values.

The BCL authors also consider 0 as an invalid handle to pass to `CloseHandle`-compatible `SafeHandle` : http://referencesource.microsoft.com/#mscorlib/microsoft/win32/safehandles/safefilehandle.cs,28